### PR TITLE
Get/set current atb

### PR DESCRIPTION
**Pull request by zachthompson**
Get and store the current atb value post-request.

@bsstoner Couple of questions.

line 89. not sure what the rationale is with this `if`.  I thought `set_atb` would get set here but it didn't.  Is this necessary?
[lines 38-53](https://github.com/duckduckgo/chrome-zeroclickinfo/blob/0883a0841da41ab0ea3dddbc1232c14e7446e87c/js/background.js#L38-L53).  Shouldn't we ditch this for a call to atb.js?

### CODE REVIEW

File: js/background.js [hunk 1]
```diff
@@ -50,9 +50,11 @@ function Background() {
             epoch = isDST ? estEpoch - oneHour : estEpoch,
             timeSinceEpoch = new Date().getTime() - epoch,
             majorVersion = Math.ceil(timeSinceEpoch / oneWeek),
-            minorVersion = Math.ceil(timeSinceEpoch % oneWeek / oneDay);
+            minorVersion = Math.ceil(timeSinceEpoch % oneWeek / oneDay), 
+            atb = 'v' + majorVersion + '-' + minorVersion;
 
-        localStorage['atb'] = 'v' + majorVersion + '-' + minorVersion;
+        localStorage['atb'] = atb;
+        localStorage['set_atb'] = atb;
```
Comment (bsstoner) [inline js/background.js]: I don't think we want to set this here. The current purpose of this field is to store when we've successfully pulled the atb param off the DOM. Once it's set we never look to the DOM again, preventing it from somehow being overwritten and changed.

**Comment by bsstoner** [comment]
For line 89 - The idea with the if statement is it's only updating ATB/variants from the DOM once, at when you install directly on the site. The check to make sure `set_atb` isn't set yet prevents the atb param + variants from changing somehow after install. The check to make sure the atb param is defined prevents us from potentially storing an empty value if there's some kind of timing issue or internal failure with placing the param on the DOM at install time. I think we need to keep it unless we change the interface with internal code. It probably didn't get set in your testing because it will only be set when you install the extension directly from duckduckgo.com.

We could probably replace line 38-53 with a call to collect.js. The only concern I have is whether the async call would introduce a race condition where we're expecting atb to be set immediately and it's not. Since it's hard to fully test this stuff end to end, it would probably be safer to get these changes out first and then come back and clean that up in a subsequent release.

File: js/background.js (hunk 1 above)
Comment (zachthompson) [inline js/background.js]: So is there any chance of it getting to onComplete and not being set?  Should we be checking set_atb there instead of just atb?

**Comment by bsstoner** [comment]
Sorry I didn't realize this was good to go. Going to run through and do a final test pass now.

File: js/background.js [hunk 2]
```diff
@@ -144,3 +144,40 @@ chrome.webRequest.onBeforeRequest.addListener(
...
+             if(curATB.version !== setATB) {
+               localStorage['set_atb'] = curATB.version;
+             }
+           }
+        }
+      };
+
+      xhr.open('GET',
+        'https://duckduckgo.com/collect.js?' + new Date().getTime()
+          + '&type=atb&atb=' + atb + '&set_atb=' + setATB,
```
Comment (bsstoner) [inline js/background.js]: @zachthompson this is a nitpick, but do you mind if I change the timestamp to a random number instead? `Math.ceil(Math.random() * 1e7)`

**Comment by bsstoner** [comment]
👍

**Pull request merged**